File: fta-backend/team/src/main/java/com/kerfaiyassine/team/services/TeamManagementService.java
package com.kerfaiyassine.team.services;

import com.kerfaiyassine.team.clients.ManagerClient;
import com.kerfaiyassine.team.clients.PlayerClient;
import com.kerfaiyassine.team.entities.Team;
import com.kerfaiyassine.team.kafka.StadiumOperationMessage;
import com.kerfaiyassine.team.repositories.TeamRepository;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Service;

@Service
public class TeamManagementService {

    private final PlayerClient playerClient;

    private final ManagerClient managerClient;

    private final KafkaTemplate<String, StadiumOperationMessage> kafkaTemplate;

    private final TeamRepository teamRepository;

    public TeamManagementService(
            PlayerClient playerClient,
            ManagerClient managerClient,
            KafkaTemplate<String, StadiumOperationMessage> kafkaTemplate,
            TeamRepository teamRepository
    ) {
        this.playerClient = playerClient;
        this.managerClient = managerClient;
        this.kafkaTemplate = kafkaTemplate;
        this.teamRepository = teamRepository;
    }

    public Object getPlayers(String teamId) {

        return playerClient.getTeamPlayers(teamId);
    }

    public Object releasePlayer(String playerId) {

        return playerClient.releasePlayer(playerId);
    }

    public Object movePlayerToYouth(String playerId) {

        return playerClient.moveToYouth(playerId);
    }

    public Object renewPlayerContract(long contractId) {

        return playerClient.renewContract(contractId);
    }

    public Object renewManagerCareer(String careerId) {

        return managerClient.renewCareer(careerId);
    }

    public Object sackManager(String careerId) {

        return managerClient.sackManager(careerId);
    }

    public String requestStadiumOperation(
            StadiumOperationMessage message
    ) {

        kafkaTemplate.send(
                "stadium-operation-topic",
                message
        );

        return "Operation request sent successfully";
    }

    public Team getTeam(String teamId) {

        return teamRepository.findById(teamId)
                .orElseThrow(
                        () -> new RuntimeException(
                                "Team not found"
                        )
                );
    }
}

File: fta-ui/src/app/Shared/services/team-management/team-management.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class TeamManagementService {

  private api =
    'http://localhost:8054/api/team-management';

  constructor(
    private http: HttpClient
  ) {}

  getTeam(
    teamId:string
  ) {

    return this.http.get(
      `${this.api}/${teamId}`
    );
  }

  getPlayers(
    teamId:string
  ) {

    return this.http.get(
      `${this.api}/${teamId}/players`
    );
  }

  releasePlayer(
    playerId:string
  ) {

    return this.http.post(
      `${this.api}/players/${playerId}/release`,
      {}
    );
  }

  moveToYouth(
    playerId:string
  ) {

    return this.http.post(
      `${this.api}/players/${playerId}/youth`,
      {}
    );
  }

  renewContract(
    contractId:number
  ) {

    return this.http.post(
      `${this.api}/contracts/${contractId}/renew`,
      {}
    );
  }

  renewCareer(
    careerId:string
  ) {

    return this.http.post(
      `${this.api}/careers/${careerId}/renew`,
      {}
    );
  }

  sackManager(
    careerId:string
  ) {

    return this.http.post(
      `${this.api}/careers/${careerId}/sack`,
      {}
    );
  }

  requestStadiumOperation(
    dto:any
  ) {

    return this.http.post(
      `${this.api}/stadium/operation`,
      dto
    );
  }

}
